### PR TITLE
Active payments count

### PR DESCRIPTION
**Pull request by adamgall**
Closes https://github.com/decentdao/decent-interface/issues/2254

So this code doesn't work.

All the typing checks out, but over in `RolePaymentDetails`, on the line where this happens:

```ts
<GreenStreamingDot isStreaming={payment.isStreaming()} />
```

at runtime `payment` doesn't have an `isStreaming()` function when a new payment is being added.

Trying to figure this out on my own right now, but @Da-Colon if you have any hints plz holler

### CODE REVIEW

File: src/components/pages/Roles/RolesTable.tsx [hunk 1]
```diff
@@ -262,7 +262,11 @@ export function RolesTable({
                 name={role.name}
                 wearerAddress={role.wearer}
                 handleRoleClick={handleRoleClick}
-                paymentsCount={role.payments?.length}
+                paymentsCount={
+                  role.payments === undefined
+                    ? undefined
+                    : role.payments.filter(p => p.isStreaming()).length
```
Comment (Da-Colon) [inline src/components/pages/Roles/RolesTable.tsx:268]: Ah good catch filtering for length

File: src/components/pages/Roles/types.tsx [hunk 2]
```diff
@@ -22,9 +22,12 @@ export interface SablierPayment extends BaseSablierStream {
   startDate: Date;
   endDate: Date;
   cliffDate: Date | undefined;
+  isStreaming: () => boolean;
 }
 
-export interface SablierPaymentFormValues extends Partial<SablierPayment> {}
+export interface SablierPaymentFormValues extends Partial<SablierPayment> {
+  isStreaming: () => boolean;
```
Comment (Da-Colon) [inline src/components/pages/Roles/types.tsx:29]: This should be on SablierPayment not the FormValues @adamgall

Comment (adamgall) [reply src/components/pages/Roles/types.tsx:29]: Ok, thank you, let me give that a shot

Comment (adamgall) [reply src/components/pages/Roles/types.tsx:29]: edit: lol well it _is_ on the `SablierPayment`, too. I added it here in order to override it so it's not optional.

File: src/hooks/DAO/loaders/useHatsTree.ts [hunk 3]
```diff
@@ -193,11 +200,24 @@ const useHatsTree = () => {
...
+                          : cliffDate !== undefined
+                            ? cliffDate.getTime()
+                            : undefined;
+                      const end = endDate ? endDate.getTime() : undefined;
+                      const cancelled = lockupLinearStream.canceled;
+                      const now = new Date().getTime();
+                      const isStreaming =
+                        !cancelled && !!start && !!end && start <= now && end > now;
+                      return isStreaming;
+                    },
```
Comment (Da-Colon) [inline src/hooks/DAO/loaders/useHatsTree.ts:220]: At this point I don't believe start and end times will ever be undefined. so probably don't need those conditionals. 

isStreaming coming out correct? hmmm

File: src/components/pages/Roles/types.tsx (hunk 2 above)
Comment (adamgall) [reply src/components/pages/Roles/types.tsx:29]: i need halp tho

File: src/hooks/DAO/loaders/useHatsTree.ts [hunk 4]
```diff
@@ -193,11 +200,24 @@ const useHatsTree = () => {
...
+                    endDate,
+                    cliffDate,
+                    isStreaming: () => {
+                      const start =
+                        lockupLinearStream.cliff === undefined &&
+                        lockupLinearStream.startTime !== undefined
+                          ? startDate.getTime()
+                          : cliffDate !== undefined
+                            ? cliffDate.getTime()
+                            : undefined;
```
Comment (DarksightKellar) [inline src/hooks/DAO/loaders/useHatsTree.ts:213]: Maybe `: cliffDate?.getTime();` for `: cliffDate.getTime() : undefined;`

File: src/hooks/DAO/loaders/useHatsTree.ts [hunk 5]
```diff
@@ -193,11 +200,24 @@ const useHatsTree = () => {
...
+                      const end = endDate ? endDate.getTime() : undefined;
```
Comment (DarksightKellar) [inline src/hooks/DAO/loaders/useHatsTree.ts:214]: `const end = endDate?.getTime()`?

**Pull request merged**